frontend/src/components/TaskView/FutureView/FutureViewDayTaskContainer.tsx

```diff
@@ -54,23 +59,59 @@ function FutureViewDayTaskContainer(
     setPrevHeights([tasksHeight, containerHeight]);
     onHeightChange(tasksHeight > containerHeight && containerHeight > 0, tasksHeight);
   });
+  const [localTasks, setLocalTasks] = useState<IdOrder[]>(idOrderList);
+  const onDragEnd = (result: DropResult): void => {
+    const { source, destination } = result;
+    if (destination == null) {
+      // invalid drop, skip
+      return;
+    }
+    const sourceOrder: number = source.index;
+    const dest = localTasks[destination.index];
+    const destinationOrder: number = dest == null ? sourceOrder : dest.order;
 
-  const taskListComponent = idOrderList.map(({ id }) => (
-    <FutureViewTask
-      key={id}
-      taskId={id}
-      containerDate={date}
-      inNDaysView={inNDaysView}
-      taskEditorPosition={taskEditorPosition}
-      calendarPosition={calendarPosition}
-      doesShowCompletedTasks={doesShowCompletedTasks}
-      isInMainList={isInMainList}
-    />
+    const reorderMap = computeReorderMap(localTasks, sourceOrder, destinationOrder);
+    setLocalTasks(getReorderedList(localTasks, reorderMap));
+    applyReorder('tasks', reorderMap);
+  };
```

Comment:
The sourceOrder value is wrong.
Compare it with
https://github.com/cornell-dti/samwise/blob/d0428006fc91b7fa1598932043aa21398975973d/frontend/src/components/TaskView/FocusView/index.tsx#L46-L69